File: app/schemas/leaderboard.py
import uuid

from pydantic import BaseModel

from app.models.user import Position


class LeaderboardEntryRead(BaseModel):
    id: uuid.UUID
    last_name: str
    first_name: str
    patronymic: str | None = None
    avatar_url: str | None = None
    position: Position | None = None
    jersey_number: int | None = None
    rating_excess: float


class LeaderboardMeRead(BaseModel):
    rank: int
    rating_excess: float

File: frontend/src/types/leaderboard.ts
import type { Position } from './user'

export interface LeaderboardEntryRead {
  id: string
  last_name: string
  first_name: string
  patronymic: string | null
  avatar_url: string | null
  position: Position | null
  jersey_number: number | null
  rating_excess: number
}

export interface LeaderboardMeRead {
  rank: number
  rating_excess: number
}
